### PR TITLE
Fix for conda compiler issues

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: bin/test_travis.sh [hunk 1]
```diff
@@ -84,7 +84,7 @@ fi
 if [[ "${BUILD_DOXYGEN}" != "" ]]; then
     cmake_line="$cmake_line -DBUILD_DOXYGEN=${BUILD_DOXYGEN}"
 fi
-if [[ "${CC}" == *"gcc"* ]] && [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
+if [[ "${CC}" == *"gcc"* ]]; then
```
Comment (certik) [inline bin/test_travis.sh]: Why is this change needed? It would be nice to test `-march=native` on Travis.

Comment (bjodah) [reply bin/test_travis.sh]: I'm guessing because of this: https://travis-ci.org/symengine/symengine/jobs/443353763#L1461

Comment (certik) [reply bin/test_travis.sh]: I see. That error typically happens when gcc generates assembly code that's for a newer processor than the one it actually runs on. This can happen if the Travis' docker container fools gcc into thinking it's on a newer processor. Ok, then we should simply not use this flag for gcc. Although I still want this to be the default, so that one gets good performance by default if you compile from source.

Comment (isuruf) [reply bin/test_travis.sh]: Looks like it was a caching issue. (We have enabled ccache and that means if the code was built previously on a different architecture, it wouldn't work.)
I've removed the commit and let's figure out the ccache issue later
